core/src/main/java/apoc/util/Utils.java

```diff
@@ -76,4 +76,15 @@ public void validate(@Name("predicate") boolean predicate, @Name("message") Stri
             throw new RuntimeException(message);
         }
     }
+
+    @UserFunction
+    @Description("apoc.util.validatePredicate(predicate, message, params) | if the predicate yields to true raise an exception else returns true, for use inside WHERE subclauses")
```

Comment:
shouldn't it fail on false? otherwise it's confusing and afaik also the opposite of the existing functionality?

Reply:
The current validate function will raise an exception if the predicate yields to true, I do find it confusing to reason about, but for consistency, I kept to this paradigm.